inc/components/class-social-share.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Class file for the Social Links component.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\Component;
+
+/**
+ * Defines the Social Links component.
+ */
+class Social_Share extends Component {
+
+	use \WP_Irving\Social;
+
+	/**
+	 * Unique component slug.
+	 *
+	 * @var string
+	 */
+	public $name = 'social-share';
+
+	/**
+	 * Define a default config shape.
+	 *
+	 * @return array Default config.
+	 */
+	public function default_config() {
+		return [
+			'post_id'        => 0,
+			'display_icons'  => true,
+		];
+	}
+
+	/**
+	 * Array of services.
+	 *
+	 * @var string
+	 */
+	public static $services = [
+		'facebook'  => [
+			'share_url' => 'https://www.facebook.com/sharer.php/',
+			'params'    => [
+				'u' => 'get_url',
+			],
+		],
+		'twitter'   => [
+			'share_url' => 'https://twitter.com/share/',
+			'params'    => [
+				'text' => 'get_title',
+				'url'  => 'get_url',
+			]
+		],
+		'whatsapp'  => [
+			'share_url' => 'https://api.whatsapp.com/send/',
+			'params'    => [
+				'text' => 'get_cta',
+			],
+		],
+		'linkedin'  => [
+			'share_url' => 'https://www.linkedin.com/shareArticle/',
+			'params'    => [
+				'url'     => 'get_url',
+				'title'   => 'get_title',
+				'summary' => 'get_excerpt',
+			],
+		],
+		'pinterest' => [
+			'share_url' => 'https://pinterest.com/pin/create/button/',
+			'params'    => [
+				'url'         => 'get_url',
+				'media'       => 'get_featured_image_url',
+				'description' => 'get_excerpt',
+			],
+		],
+	];
+
+	/**
+	 * Add Social_Item component for each sharing service.
+	 *
+	 * @param WP_Post $post Post object these sharing buttons will share.
+	 * @return Social_Share Return this class.
+	 */
+	public function add_links_for_post( $post ) : self {
```

Comment:
```suggestion
	public function add_links_for_post( \WP_Post $post ) : self {
```

Reply:
This way we don't have to validate $post in the function, we know it'll be valid.